### PR TITLE
[IIO] Custom channel index number as input

### PR DESCRIPTION
**Pull request by kparichay**
Added the option for user to specify custom channel number as input
to enable specific channels.
Added test case for custom channel index as well.

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [ ]Skipped
2. Run test: [ ]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 1]
```diff
@@ -1070,6 +1072,30 @@ gst_tensor_src_iio_set_property (GObject * object, guint prop_id,
         self->channels_enabled = CHANNELS_ENABLED_ALL;
       } else if (!g_strcmp0 (param, CHANNELS_ENABLED_AUTO_CHAR)) {
         self->channels_enabled = CHANNELS_ENABLED_AUTO;
+      } else {
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: This is not related with this PR, so I wouldn't apply it to this PR.
However, for property decision, please use case insensitive mechanisms instead of g_strcmp0. Glib has case insensitive strcmp API.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 2]
```diff
@@ -399,6 +400,7 @@ gst_tensor_src_iio_init (GstTensorSrcIIO * self)
   /** init properties */
   self->configured = FALSE;
   self->channels = DEFAULT_PROP_STRING;
+  self->custom_channel_list = DEFAULT_PROP_STRING;
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: ```custom_channel_list``` is GList. It does not incur erros because ```DEFAULT_PROP_STRING``` is NULL; however, it may mislead other developers to update ```DEFAULT_PROP_STRING``` with real string values, which will break it. It's better be written NULL instead of DEFAULT_PROP_STRING.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 1 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Noted, will upload a new PR for case insensitive comparison.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 3]
```diff
@@ -1619,6 +1650,34 @@ gst_tensor_src_iio_setup_scan_channels (GstTensorSrcIIO * self)
     }
   }
 
+  /** enable the custom channels and disable the rest */
+  if (self->channels_enabled == CHANNELS_ENABLED_CUSTOM) {
+    for (ch_list = self->channels; ch_list != NULL; ch_list = ch_list->next) {
+      channel_prop = (GstTensorSrcIIOChannelProperties *) ch_list->data;
+      item =
+          g_list_find (self->custom_channel_list,
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: Could it be possible to use more efficient mechanism?
Searching for a linked list for every ch_list appears to be less efficient.

If the number of channels is limited to 32 or 64, you may simply use bitmap with uint32 or uint64, simplifying ```_GstTensorSrcIIO``` as well.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 2 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Updated.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 3 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: I cannot find any upper limit on the the number of channels for a device.

What do you think about using GArray instead of GList, sort the array (`custom_channel_list`) and then use binary search (binary search will have to be implemented).

Comment (myungjoo) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: https://developer.gnome.org/glib/stable/glib-Hash-Tables.html may be fast enough. You may use this as a key-value storage without values.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 4]
```diff
@@ -314,7 +314,8 @@ gst_tensor_src_iio_class_init (GstTensorSrcIIOClass * klass)
       g_param_spec_string ("channels", "Channels to be enabled",
           "Enable channels -"
           "auto: enable all channels when no channels are enabled automatically"
-          "all: enable all channels",
+          "all: enable all channels"
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: Later, please consider to show proper description using gst-inspect.
now $ gst-inspect-1.0 tensor_src_iio
```
$ gst-inspect-1.0 tensor_src_iio
..
  channels            : Enable channels -auto: enable all channels when no channels are enabled automaticallyall: enable all channels
```

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 3 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: updated using hashtable.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 4 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Updated.

File: gst/nnstreamer/tensor_source/tensor_src_iio.c (hunk 1 above)
Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Added #1360 .
